### ISSUE TITLE
4.3.1 / gfx803 / tensorflow-rocm 2.6.0 - librocblas.so.0: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by sos-michael** [RadeonOpenCompute/ROCm#1578]
So i know gfx803 is only unofficially support now, but any help would be great!

Ubuntu 20.04
gfx803
epyc zen1

so I just run:
`sudo apt install rocm-dkms rocm-libs`

rocm-smi shows all my gpus (all are the same gfx803)

and install tensorflow-rocm into a virtualenv from inside of pycharm

when i try to run:

`from ai_benchmark import AIBenchmark`
`benchmark = AIBenchmark(use_CPU=False, verbose_level=1)`
`results = benchmark.run()`

I get the following error:

`~/Projects/mlenv/lib/python3.8/site-packages/tensorflow/python/pywrap_tensorflow.py in <module>`
`     63   try:`
`---> 64     from tensorflow.python._pywrap_tensorflow_internal import *`
`     65   # This try catch logic is because there is no bazel equivalent for py_extension.`

`ImportError: librocblas.so.0: cannot open shared object file: No such file or directory`

If install this dirty version of rocblas from here(which is for 4.3.0 not 4.3.1):
https://github.com/xuhuisheng/rocm-gfx803

the benchmark runs, but doesn't see my GPUs and rocm-smi show 0% activity...

**Comment by xuhuisheng** [comment]
tensorflow_rocm-2.6.0 drop gfx803 support.
You could compile tensorflow_rocm-2.6.0 by yourself.

In my environment, tensorflow_rocm-2.6.0 can run properly on gfx803.

```
work@91f6c555a036:~/test$ python3 test.py
2021-09-25 07:34:07.692003: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-09-25 07:34:07.693618: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 7692 MB memory:  -> device: 0, name: Device 67df, pci bus id: 0000:02:00.0
2021-09-25 07:34:08.772256: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)
2021-09-25 07:34:08.798666: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
2021-09-25 07:34:08.804292: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
2021-09-25 07:34:08.807748: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
Epoch 1/5
2021-09-25 07:34:09.096113: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
1875/1875 [==============================] - 5s 2ms/step - loss: 0.3016 - accuracy: 0.9119
Epoch 2/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1476 - accuracy: 0.9551
Epoch 3/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1100 - accuracy: 0.9663
Epoch 4/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.0891 - accuracy: 0.9732
Epoch 5/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.0772 - accuracy: 0.9758
2021-09-25 07:34:31.313063: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
2021-09-25 07:34:31.318229: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
2021-09-25 07:34:31.322008: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
2021-09-25 07:34:31.445636: I tensorflow/core/common_runtime/gpu_fusion_pass.cc:507] ROCm Fusion is enabled.
313/313 [==============================] - 1s 2ms/step - loss: 0.0735 - accuracy: 0.9777

```

**Comment by sos-michael** [comment]
That is awesome to know! So if I just stick with your version of rocblas and an older version (~2.4)  of tensorflow-rocm I should be okay? Do you have any plans of packaging up your versions of tensorflow-rocm as well?

**Comment by xuhuisheng** [comment]
The tensorflow_rocm-2.4.3 had official support on gfx803.

I just upload gfx803 version of tensorflow_rocm-2.6.0 to <https://github.com/xuhuisheng/rocm-gfx803/releases/download/rocm43/tensorflow_rocm-2.6.0-cp38-cp38-linux_x86_64.whl>, please try it.

The source codes comes from this branch: <https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/tree/r2.6-rocm-enhanced>

**Comment by sos-michael** [comment]
pip3 install tensorflow_rocm-2.6.0-cp38-cp38-linux_x86_64.whl 
Processing ./tensorflow_rocm-2.6.0-cp38-cp38-linux_x86_64.whl
ERROR: tensorflow-rocm has an invalid wheel, .dist-info directory 'tensorflow-2.6.0.dist-info' does not start with 'tensorflow-rocm'

**Comment by xuhuisheng** [comment]
It's wiered that I didn't meet this error.
The original package is tensorflow-2.6.0-cp38-cp38-linux_x86_64.whl, I renamed it to tensorflow_rocm-2.6.0-cp38-cp38-linux_x86_64.whl manually.

**Comment by sos-michael** [comment]
It installed! But I appear to be no closer to my destination. My apologies if I am doing something daft :/.

So to give myself a fresh start I did:

`sudo apt purge roc* hip* hsa* llvm-amdgpu comgr gcc-7-base gcc-8-base hsa-amd-aqlprofile libasan4 libbabeltrace-ctf1 libbabeltrace1 libcilkrts5 libdrm-dev libdw1 libelf-dev libfile-copy-recursive-perl libfile-which-perl libgcc-7-dev libgl-dev libglx-dev libmpx2 libncurses5 libpthread-stubs0-dev libstdc++-7-dev libtinfo5 libubsan0 liburi-encode-perl libx11-dev libxau-dev libxcb1-dev libxdmcp-dev mesa-common-dev openmp-extras x11proto-core-dev x11proto-dev xorg-sgml-doctools xtrans-dev dctrl-tools dkms`


`sudo rm -r /opt/rocm*`
**reboot**

`sudo apt install rocm-dkms`
**reboot**

`sudo apt install rocm-libs`
`sudo apt install ./rocblas_2.39.0-8328dcce.dirty_amd64.deb`

`source Projects/rocm-ml/bin/activate`
`pip3 install tensorflow-2.6.0-cp38-cp38-linux_x86_64.whl`
`pip3 install ai-benchmark`

then when I run the AI benchmarks I see this (not that it doesn't mention any GPU)
`from ai_benchmark import AIBenchmark`
`benchmark = AIBenchmark(use_CPU=False, verbose_level=1)`
`results = benchmark.run()`

>   AI-Benchmark-v.0.1.2   
>   Let the AI Games begin..

> *  TF Version: 2.6.0
> *  Platform: Linux-5.11.0-36-generic-x86_64-with-glibc2.29
> *  CPU: N/A
> *  CPU RAM: 252 GB

> The benchmark is running...
> The tests might take up to 20 minutes
> Please don't interrupt the script

> 1/19. MobileNet-V2

> 1.1 - inference | batch=50, size=224x224: 347 ± 61 ms
> 1.2 - training  | batch=50, size=224x224: 1629 ± 196 ms

while rocm-smi looks like:

> ======================= ROCm System Management Interface =======================
> ================================= Concise Info =================================
> GPU  Temp   AvgPwr  SCLK     MCLK      Fan       Perf   PwrCap  VRAM%  GPU%  
> 0        38.0c   9.1W      300Mhz  300Mhz  18.82%  auto  98.0W     0%          0%

**Comment by xuhuisheng** [comment]
Is is correct?


```
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from ai_benchmark import AIBenchmark
>>> benchmark = AIBenchmark(use_CPU=False, verbose_level=1)

>>   AI-Benchmark-v.0.1.2   
>>   Let the AI Games begin..

>>> results = benchmark.run()
*  TF Version: 2.6.0
*  Platform: Linux-5.11.0-27-generic-x86_64-with-glibc2.29
*  CPU: N/A
*  CPU RAM: 47 GB
*  GPU/0: Device 67df
*  GPU RAM: 7.5 GB
*  CUDA Version: N/A
*  CUDA Build: N/A


The benchmark is running...
The tests might take up to 20 minutes
Please don't interrupt the script

1/19. MobileNet-V2

1.1 - inference | batch=50, size=224x224: 99.6 ± 9.2 ms
1.2 - training  | batch=50, size=224x224: 4491 ± 71 ms

2/19. Inception-V3

2.1 - inference | batch=20, size=346x346: 145 ± 1 ms

```

`watch -d /opt/rocm/bin/rocm-smi`

```
======================= ROCm System Management Interface =======================
================================= Concise Info =================================
GPU  Temp   AvgPwr   SCLK     MCLK    Fan     Perf  PwrCap  VRAM%  GPU%  
0    41.0c  22.083W  1300Mhz  300Mhz  28.63%  auto  145.0W   95%   0%    
================================================================================
============================= End of ROCm SMI Log ==============================
```

**Comment by sos-michael** [comment]
That is 100% the correct output. I'm glad to know it at least can work!

**Comment by xuhuisheng** [comment]
I just run a small script, and the log shows the cpu infos

```
>>> import tensorflow as tf
>>> a = tf.constant([1.0, 2.0, 3.0], shape=[3], name='a')
2021-09-26 13:40:11.230102: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-09-26 13:40:11.233077: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 7692 MB memory:  -> device: 0, name: Device 67df, pci bus id: 0000:02:00.0
>>> b = tf.constant([1.0, 2.0, 3.0], shape=[3], name='b')
>>> c = a + b

```

I think you could test some small examples, likes square demo of HIP. to make sure the ROCm had installed properly.
<https://github.com/xuhuisheng/rocm-build/blob/master/check/run-hip.sh>

**Comment by sos-michael** [comment]
Well this great(-ish) news. So something is very wrong with my setup! 

But as far as I can tell everything is setup correctly. What I said above is exactly, to the letter, what I did. I'm going to have to do some digging. I also have an nvidia GPU in this setup, there is nothing I can tell I did differently. 
 
./run-hip.sh 
error: 'hipErrorInvalidDevice'(101) at src/test_hip.cpp:35


when i run rocminfo I see this:
```
*******                  
Agent 5                  
*******                  
  Name:                    gfx803                             
  Uuid:                    GPU-XX                             
  Marketing Name:          Ellesmere [Radeon Pro WX 7100]     
  Vendor Name:             AMD                                
  Feature:                 KERNEL_DISPATCH                    
  Profile:                 BASE_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        128(0x80)                          
  Queue Min Size:          4096(0x1000)                       
  Queue Max Size:          131072(0x20000)                    
  Queue Type:              MULTI                              
  Node:                    4                                  
  Device Type:             GPU                                
  Cache Info:              
    L1:                      16(0x10) KB                        
  Chip ID:                 26564(0x67c4)                      
  Cacheline Size:          64(0x40)                           
  Max Clock Freq. (MHz):   1243                               
  BDFID:                   768                                
  Internal Node ID:        4                                  
  Compute Unit:            36                                 
  SIMDs per CU:            4                                  
  Shader Engines:          4                                  
  Shader Arrs. per Eng.:   1                                  
  WatchPts on Addr. Ranges:4                                  
  Features:                KERNEL_DISPATCH 
  Fast F16 Operation:      FALSE                              
  Wavefront Size:          64(0x40)                           
  Workgroup Max Size:      1024(0x400)                        
  Workgroup Max Size per Dimension:
    x                        1024(0x400)                        
    y                        1024(0x400)                        
    z                        1024(0x400)                        
  Max Waves Per CU:        40(0x28)                           
  Max Work-item Per CU:    2560(0xa00)                        
  Grid Max Size:           4294967295(0xffffffff)             
  Grid Max Size per Dimension:
    x                        4294967295(0xffffffff)             
    y                        4294967295(0xffffffff)             
    z                        4294967295(0xffffffff)             
  Max fbarriers/Workgrp:   32                                 
  Pool Info:               
    Pool 1                   
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED      
      Size:                    16777216(0x1000000) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       FALSE                              
    Pool 2                   
      Segment:                 GROUP                              
      Size:                    64(0x40) KB                        
      Allocatable:             FALSE                              
      Alloc Granule:           0KB                                
      Alloc Alignment:         0KB                                
      Accessible by all:       FALSE                              
  ISA Info:                
    ISA 1                    
      Name:                    amdgcn-amd-amdhsa--gfx803          
      Machine Models:          HSA_MACHINE_MODEL_LARGE            
      Profiles:                HSA_PROFILE_BASE                   
      Default Rounding Mode:   NEAR                               
      Default Rounding Mode:   NEAR                               
      Fast f16:                TRUE                               
      Workgroup Max Size:      1024(0x400)                        
      Workgroup Max Size per Dimension:
        x                        1024(0x400)                        
        y                        1024(0x400)                        
        z                        1024(0x400)                        
      Grid Max Size:           4294967295(0xffffffff)             
      Grid Max Size per Dimension:
        x                        4294967295(0xffffffff)             
        y                        4294967295(0xffffffff)             
        z                        4294967295(0xffffffff)             
      FBarrier Max Size:       32   
```

**Comment by xuhuisheng** [comment]
Change AMD_LOG_LEVEL=0 to AMD_LOG_LEVEL=6, it will print more logs.

```
work@c9d2351c8df1:~/rocm-build/check$ bash run-hip.sh 
:3:rocdevice.cpp            :430 : 2249897949801 us: Initializing HSA stack.
:3:comgrctx.cpp             :33  : 2249897987271 us: Loading COMGR library.
:3:rocdevice.cpp            :196 : 2249897994495 us: Numa selects cpu agent[0]=0x9732a0(fine=0xa0b680,coarse=0xa0c540, kern_arg=0xa30fb0) for gpu agent=0x7f6343007a84
:3:rocdevice.cpp            :1562: 2249897994952 us: HMM support: 0, xnack: 0

:4:rocdevice.cpp            :1858: 2249897995046 us: Allocate hsa host memory 0x1012000, size 0x28
:4:rocdevice.cpp            :1858: 2249897995578 us: Allocate hsa host memory 0x1100dff000, size 0x101000
:4:rocdevice.cpp            :1858: 2249897996116 us: Allocate hsa host memory 0x11010ff000, size 0x101000
:4:runtime.cpp              :82  : 2249897996264 us: init
:3:hip_device.cpp           :239 : 2249897996311 us: 182280: [7f634262e880] hipGetDeviceProperties: Returned hipSuccess : 
info: running on device Device 67df
info: allocate host mem (  7.63 MB)
info: allocate device mem (  7.63 MB)
:3:hip_memory.cpp           :384 : 2249898000349 us: 182280: [7f634262e880] hipMalloc ( 0x7fffab1166b0, 4000000 )
:4:rocdevice.cpp            :1993: 2249898000551 us: Allocate hsa device memory 0x1101600000, size 0x3d0900
:3:rocdevice.cpp            :2032: 2249898000560 us: device=0xa43920, freeMem_ = 0xffc2f700
:3:hip_memory.cpp           :386 : 2249898000575 us: 182280: [7f634262e880] hipMalloc: Returned hipSuccess : 0x1101600000: duration: 226 us
:3:hip_memory.cpp           :384 : 2249898000583 us: 182280: [7f634262e880] hipMalloc ( 0x7fffab1166a8, 4000000 )
:4:rocdevice.cpp            :1993: 2249898000738 us: Allocate hsa device memory 0x1101e00000, size 0x3d0900
:3:rocdevice.cpp            :2032: 2249898000746 us: device=0xa43920, freeMem_ = 0xff85ee00
:3:hip_memory.cpp           :386 : 2249898000760 us: 182280: [7f634262e880] hipMalloc: Returned hipSuccess : 0x1101e00000: duration: 177 us
info: copy Host2Device
:3:hip_memory.cpp           :429 : 2249898000786 us: 182280: [7f634262e880] hipMemcpy ( 0x1101600000, 0x7f6341a5c010, 4000000, hipMemcpyHostToDevice )
:3:rocdevice.cpp            :2543: 2249898001106 us: number of allocated hardware queues with low priority: 0, with normal priority: 0, with high priority: 0, maximum per priority is: 4
:3:rocdevice.cpp            :2618: 2249898015577 us: created hardware queue 0x1024000 with size 1024 with priority 1, cooperative: 0
:4:rocdevice.cpp            :1858: 2249898015805 us: Allocate hsa host memory 0x1100d00000, size 0x80000
:3:devprogram.cpp           :2466: 2249898424507 us: Using Code Object V4.
:4:command.cpp              :303 : 2249898426431 us: command is enqueued: 0xa714d0
:4:command.cpp              :262 : 2249898426459 us: queue marker to command queue: 0xa76dd0
:4:commandqueue.cpp         :176 : 2249898426461 us: command (CopyHostToDevice) is submitted: 0xa714d0
:4:command.cpp              :303 : 2249898426472 us: command is enqueued: 0x973fe0
:4:command.cpp              :222 : 2249898426504 us: waiting for event 0xa714d0 to complete, current status 2
:4:rocvirtual.hpp           :200 : 2249898426678 us: [7f6340e89640]!	 WaitCurret completion_signal=0x1001b00
:4:rocvirtual.hpp           :228 : 2249898426696 us: [7f6340e89640]!	 WaitNext completion_signal=0x1001a80
:4:rocblit.cpp              :670 : 2249898426710 us: [7f6340e89640]!	 HSA Asycn Copy wait_event=0x0, completion_signal=0x1001b00
:4:rocvirtual.hpp           :200 : 2249898438108 us: [7f6340e89640]!	 WaitCurret completion_signal=0x1001b00
:4:rocvirtual.cpp           :449 : 2249898438123 us: [7f6340e89640]!	 Host wait on completion_signal=0x1001b00
:4:commandqueue.cpp         :176 : 2249898438609 us: command (InternalMarker) is submitted: 0x973fe0
:4:rocvirtual.hpp           :200 : 2249898438618 us: [7f6340e89640]!	 WaitCurret completion_signal=0x1001b00
:4:command.cpp              :152 : 2249898438627 us: Command 0xa714d0 complete
:4:command.cpp              :236 : 2249898438628 us: event 0xa714d0 wait completed
:4:command.cpp              :152 : 2249898438633 us: Command 0x973fe0 complete
:3:hip_memory.cpp           :432 : 2249898438645 us: 182280: [7f634262e880] hipMemcpy: Returned hipSuccess : : duration: 437859 us
info: launch 'vector_square' kernel
:3:hip_platform.cpp         :202 : 2249898438702 us: 182280: [7f634262e880] __hipPushCallConfiguration ( {512,1,1}, {256,1,1}, 0, stream:<null> )
:3:hip_platform.cpp         :206 : 2249898438719 us: 182280: [7f634262e880] __hipPushCallConfiguration: Returned hipSuccess : 
:3:hip_platform.cpp         :213 : 2249898438737 us: 182280: [7f634262e880] __hipPopCallConfiguration ( {4227128,0,1}, {1139855620,32611,1}, 0x7fffab1166c0, 0x7fffab1166b8 )
:3:hip_platform.cpp         :222 : 2249898438752 us: 182280: [7f634262e880] __hipPopCallConfiguration: Returned hipSuccess : 
:3:hip_module.cpp           :489 : 2249898438769 us: 182280: [7f634262e880] hipLaunchKernel ( 0x401c10, {512,1,1}, {256,1,1}, 0x7fffab116700, 0, stream:<null> )
:3:devprogram.cpp           :2466: 2249898439108 us: Using Code Object V4.
:3:hip_module.cpp           :358 : 2249898439536 us: 182280: [7f634262e880] ihipModuleLaunchKernel ( 0x0xa783f0, 131072, 1, 1, 256, 1, 1, 0, stream:<null>, 0x7fffab116700, char array:<null>, event:0, event:0, 0, 0 )
:4:command.cpp              :303 : 2249898439559 us: command is enqueued: 0xa7d2d0
:3:hip_platform.cpp         :638 : 2249898439571 us: 182280: [7f634262e880] ihipLaunchKernel: Returned hipSuccess : 
:3:hip_module.cpp           :491 : 2249898439579 us: 182280: [7f634262e880] hipLaunchKernel: Returned hipSuccess : 
info: copy Device2Host
:4:commandqueue.cpp         :176 : 2249898439574 us: command (KernelExecution) is submitted: 0xa7d2d0
:3:hip_memory.cpp           :429 : 2249898439595 us: 182280: [7f634262e880] hipMemcpy ( 0x7f634168b010, 0x1101e00000, 4000000, hipMemcpyDeviceToHost )
:4:command.cpp              :303 : 2249898439602 us: command is enqueued: 0xa714d0
:3:rocvirtual.cpp           :603 : 2249898439603 us: !	arg0:  C_d.coerce = ptr:0x1101e00000 obj:[0x1101e00000-0x11021d0900] threadId : 7f6340e89640
:3:rocvirtual.cpp           :603 : 2249898439619 us: !	arg1:  A_d.coerce = ptr:0x1101600000 obj:[0x1101600000-0x11019d0900] threadId : 7f6340e89640
:4:command.cpp              :262 : 2249898439612 us: queue marker to command queue: 0xa76dd0
:4:command.cpp              :303 : 2249898439635 us: command is enqueued: 0x973910
:3:rocvirtual.cpp           :2560: 2249898439628 us: [7f6340e89640]!	ShaderName : _Z13vector_squareIfEvPT_S1_m
:4:command.cpp              :222 : 2249898439645 us: waiting for event 0xa714d0 to complete, current status 3
:4:rocvirtual.cpp           :753 : 2249898439667 us: [7f6340e89640] HWq=0x1024000, Dispatch Header = 0x502 (type=2, barrier=1, acquire=2, release=0), setup=3, grid=[131072, 1, 1], workgroup=[256, 1, 1], private_seg_size=0, group_seg_size=0, kernel_obj=0x1102608840, kernarg_address=0x1100d00000, completion_signal=0x0
:4:commandqueue.cpp         :176 : 2249898439679 us: command (CopyDeviceToHost) is submitted: 0xa714d0
:4:rocvirtual.hpp           :200 : 2249898442945 us: [7f6340e89640]!	 WaitCurret completion_signal=0x1001a80
:4:rocvirtual.hpp           :228 : 2249898442955 us: [7f6340e89640]!	 WaitNext completion_signal=0x1001a00
:4:rocvirtual.cpp           :871 : 2249898442964 us: [7f6340e89640] HWq=0x1024000, BarrierAND Header = 0x1503 (type=3, barrier=1, acquire=2, release=2), dep_signal=[0x0, 0x0, 0x0, 0x0, 0x0], completion_signal=0x1001a80
:4:rocvirtual.hpp           :200 : 2249898442977 us: [7f6340e89640]!	 WaitCurret completion_signal=0x1001a00
:4:rocvirtual.hpp           :228 : 2249898442987 us: [7f6340e89640]!	 WaitNext completion_signal=0x1001980
:4:rocblit.cpp              :670 : 2249898442995 us: [7f6340e89640]!	 HSA Asycn Copy wait_event=0x0, completion_signal=0x1001a00
:4:rocvirtual.hpp           :200 : 2249898443559 us: [7f6340e89640]!	 WaitCurret completion_signal=0x1001a00
:4:rocvirtual.cpp           :449 : 2249898443568 us: [7f6340e89640]!	 Host wait on completion_signal=0x1001a00
:4:commandqueue.cpp         :176 : 2249898443971 us: command (InternalMarker) is submitted: 0x973910
:4:rocvirtual.hpp           :200 : 2249898443978 us: [7f6340e89640]!	 WaitCurret completion_signal=0x1001a00
:4:command.cpp              :152 : 2249898443985 us: Command 0xa7d2d0 complete
:4:command.cpp              :152 : 2249898443993 us: Command 0xa714d0 complete
:4:command.cpp              :152 : 2249898444002 us: Command 0x973910 complete
:4:command.cpp              :236 : 2249898443995 us: event 0xa714d0 wait completed
:3:hip_memory.cpp           :432 : 2249898444034 us: 182280: [7f634262e880] hipMemcpy: Returned hipSuccess : : duration: 4439 us
info: check result
PASSED!

```

**Comment by ROCmSupport** [comment]
Thanks @sos-michael for reaching out.
ROCm does not support gfx8 officially and so no official help comes from ROCm team.
Request to follow suggestions from active supporting users like @xuhuisheng 
Thank you.

**Comment by sos-michael** [comment]
They are actually RPDs (I've got 5 of the them), but as far as I know the RPD/wx7100 is just a Polaris/rx580 down-clocked. I've used rocm on it for years. 

./run-hip.sh 
:3:rocdevice.cpp            :430 : 47612163809 us: Initializing HSA stack.
:1:rocdevice.cpp            :438 : 47612163925 us: hsa_init failed.
:4:runtime.cpp              :82  : 47612163942 us: init
:3:hip_device.cpp           :148 : 47612163950 us: 9380 : [7f9bfd9a7880] hipGetDeviceProperties: Returned hipErrorInvalidDevice : 
error: 'hipErrorInvalidDevice'(101) at src/test_hip.cpp:35

I am pretty confident that i am just missing something.

**Comment by xuhuisheng** [comment]
`:1:rocdevice.cpp :438 : 47612163925 us: hsa_init failed.`

It is said that device init failed.
<https://github.com/ROCm-Developer-Tools/ROCclr/blob/rocm-4.3.x/device/rocm/rocdevice.cpp#L438>

hsa_init() just try to acquire the Runtime, but failed.
<https://github.com/RadeonOpenCompute/ROCR-Runtime/blob/rocm-4.3.x/src/core/runtime/hsa.cpp#L206>
<https://hub.fastgit.org/RadeonOpenCompute/ROCR-Runtime/blob/rocm-4.3.x/src/core/runtime/runtime.cpp#L94>

Guess the reason is HSA_STATUS_ERROR_OUT_OF_RESOURCES.
Need recompile ROCR-Runtime, print more log.

**Comment by sos-michael** [comment]
@xuhuisheng ,

I just wanted to take a minute and say thank you! I totally got this to work with a bit more futzing and it would not have happened without your help.
